tests/unit/test_cli.py

```diff
@@ -21,3 +27,56 @@ def test_cli_main(self, mock_exporter):
         config = Config(port=mock_port, level=mock_level, enable_collectors=mock_enable_collectors)
         main(config)
         mock_exporter.assert_called_once_with(mock_port)
+
+    @pytest.mark.parametrize(
+        "enable_collectors",
+        [
+            (["collector.mega_raid"]),
+            (["collector.ipmi_dcmi", "collector.ipmi_sel", "collector.ipmi_sensor"]),
+            (["collector.lsi_sas_2", "collector.lsi_sas_3"]),
+            (["collector.mega_raid", "collector.poweredge_raid"]),
+        ],
+    )
+    @patch.object(__main__, "Exporter")
+    def test_cli_main_enable_collector(self, mock_exporter, enable_collectors):
+        config = Config(port=10000, level="INFO", enable_collectors=enable_collectors)
+
+        collector_registries = get_collector_registries(config)
+
+        for collector in enable_collectors:
+            if collector in collector_registries:
+                collector_registries[collector] = Mock()
+
+        with patch.object(__main__, "get_collector_registries", return_value=collector_registries):
+            main(config)
+
+        for collector in enable_collectors:
+            mock_exporter.return_value.register.assert_any_call(
+                collector_registries.get(collector)
+            )
+
+        for name, collector in collector_registries.items():
+            if name.lower() not in enable_collectors:
+                with pytest.raises(AssertionError):
+                    mock_exporter.return_value.register.assert_called_with(collector)
```

Comment:
I think that you should mock `get_collector_registries` only and not run it. Maybe you can provide list of collectors in constant and use it also in `test_get_collector_registries`.
```suggestion
    def test_cli_main_enable_collector(self, mock_exporter, enable_collectors):
        config = Config(port=10000, level="INFO", enable_collectors=enable_collectors)
        collector_registries = {
            "collector.hpe_ssa": "collector.hpe_ssa",
            "collector.ipmi_dcmi": "collector.ipmi_dcmi",
            "collector.ipmi_sel": "collector.ipmi_sel",
            "collector.ipmi_sensor": "collector.ipmi_sensor",
            "collector.lsi_sas_2": "collector.lsi_sas_2",
            "collector.lsi_sas_3": "collector.lsi_sas_3",
            "collector.mega_raid": "collector.mega_raid",
            "collector.poweredge_raid": "collector.poweredge_raid",
            "collector.redfish": "collector.redfish",
        }
        
        with patch.object(__main__, "get_collector_registries", return_value=collector_registries):
            main(config)

        mock_exporter.register.assert_has_calls(
            [call(collector) for collector in enable_collectors], any_order=True,
        )
```

Reply:
There is not side effect on `get_collector_registries`, all the collector initialize is clean. I don't think it's that necessary to avoid the call.

Reply:
I'm okay with James' solution. `get_collector_registries` is not the key point here and the correctness will be verified in the next unit test function.

Reply:
Robert's solution is good. But I'm okay with James' solution too. `get_collector_registries` is not the key point here and the correctness will be verified by next unit test function.

Reply:
In general you should mocked any other functions to test single function with single test, but in here you are right it's not necessary. That's why I put only comment and not request changes.